graphql/schemabuilder/pagination.go

```diff
@@ -372,6 +380,62 @@ func (c *connectionContext) pagesFromEdges(edges []Edge, limit int) (pages []str
 	return pages
 }
 
+var matchGroups = regexp.MustCompile(`(?:([^\s"]+)|"([^"]*)"?)+`)
+
+// Split terms by spaces, except for spaces within quotes.
+// Like http://stackoverflow.com/questions/16261635/javascript-split-string-by-space-but-ignore-space-in-quotes-notice-not-to-spli
+// except:
+// - Treat the query ["san fran] as having the single term ["san fran"] instead of ["san", "fran"]
+// - Removes the delimiting quotes.
+func matchString(str, query string) bool {
+	matches := matchGroups.FindAllStringSubmatch(query, -1)
+	for _, match := range matches {
+		if len(match) < 2 {
+			continue
+		}
+		if strings.Contains(str, match[1]) {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *connectionContext) applyTextFilter(nodes []interface{}, args PaginationArgs) ([]interface{}, error) {
+	if args.Filter == nil || *args.Filter == "" {
+		return nodes, nil
+	}
+
+	var filteredNodes []interface{}
+	for _, node := range nodes {
+		keep := false
+		// For each possible field we're matching against, check if there's a match.
+		for _, filterField := range c.FilterTextFields {
+			// Resolve the graphql.Field made for sorting.
+			text, err := filterField.Resolve(context.TODO(), node, nil, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			// Only strings are allowed for FilterText fields.
+			textString, ok := text.(string)
+			if !ok {
+				return nil, errors.New("Text was not a string")
```

Comment:
include the name of the node here?  Or some other identifier to indicate that a filter on field "Blah" was not valid because it's not a string.

Reply:
That's fair - I can provide the name of the text filter. It can be something like:
```go
"filter %s returned %T, must be a string", name, text
```